test/integration/IntegrationTest.h

```diff
@@ -36,3 +36,100 @@ struct IntegrationTest : public ::testing::Test
 
     void initBridge(config::Config& config);
 };
+
+template <typename T>
+class GroupCall
+{
+public:
+    GroupCall(const std::vector<T*> clients) : _clients(clients) {}
+
+    bool connect(uint64_t timeout)
+    {
+        auto start = utils::Time::getAbsoluteTime();
+        for (auto client : _clients)
+        {
+            if (!client->_channel.isSuccess())
+            {
+                return false;
+            }
+        }
+
+        for (auto client : _clients)
+        {
+            client->processOffer();
+            if (!client->_transport || !client->_audioSource)
+            {
+                return false;
+            }
+        }
+
+        for (auto client : _clients)
+        {
+            client->connect();
+        }
+
+        for (size_t connectedCount = 0; utils::Time::getAbsoluteTime() - start < timeout; connectedCount = 0)
+        {
+            for (auto client : _clients)
+            {
+                if (!client->_transport->isConnected())
+                {
+                    break;
+                }
+                else
+                {
+                    ++connectedCount;
+                }
+            }
+
+            if (connectedCount == _clients.size())
+            {
+                return true;
+            }
+            utils::Time::nanoSleep(1 * utils::Time::sec);
+            logger::debug("waiting for connect...", "test");
+        }
```

Comment:
I think this can be simplified, like:
```
while (utils::Time::getAbsoluteTime() - start < timeout)
{
    auto it = std::find_if_not(_clients.begin(), _clients.end(), 
        [] (auto c) { return c->_transport->isConnected(); });
        
    if (it == _clients.end())
    {
         return true;
    }
    utils::Time::nanoSleep(1 * utils::Time::sec);
    logger::debug("waiting for connect...", "test");
}
```